Description: <Add sunway support>
Author: Miao Changwei <miaochangwei@uniontech.com>

---
Last-Update: 2022-04-17

--- libev-4.33.orig/ev.c
+++ libev-4.33/ev.c
@@ -493,7 +493,7 @@
 
 #if EV_USE_IOURING
 # include <sys/syscall.h>
-# if !SYS_io_uring_setup && __linux && !__alpha
+# if !SYS_io_uring_setup && __linux && !__alpha && !__sw_64
 #  define SYS_io_uring_setup     425
 #  define SYS_io_uring_enter     426
 #  define SYS_io_uring_wregister 427
@@ -839,6 +839,8 @@ struct signalfd_siginfo
       #define ECB_MEMORY_FENCE         __asm__ __volatile__ ("tb1 0,%%r0,128" : : : "memory")
     #elif defined __sh__
       #define ECB_MEMORY_FENCE         __asm__ __volatile__ (""         : : : "memory")
+    #elif defined __sw_64__
+      #define ECB_MEMORY_FENCE         __asm__ __volatile__ ("memb"       : : : "memory")
     #endif
   #endif
 #endif
@@ -1555,6 +1557,7 @@ ecb_binary32_to_binary16 (uint32_t x)
     || defined __m68k__ \
     || defined __m88k__ \
     || defined __sh__ \
+    || defined __sw_64__ \
     || defined _M_IX86 || defined ECB_MSVC_AMD64 || defined _M_IA64 \
     || (defined __arm__ && (defined __ARM_EABI__ || defined __EABI__ || defined __VFP_FP__ || defined _WIN32_WCE || defined __ANDROID__)) \
     || defined __aarch64__
